src/Refinery/String/MakeClickable.php

```diff
@@ -38,15 +58,22 @@ public function transform($from) : string
         $endOfMatch = 0;
         $stringParts = [];
         $matches = [];
-        while (1 === preg_match('@(^|[^[:alnum:]])(((https?://)|(www.))[^[:cntrl:][:space:]<>\'"]+)([^[:alnum:]]|$)@', substr($from, $endOfMatch), $matches)) {
+        [$matcher, $pattern, $check] = $this->determineParameters();
+        
+        while ($check == call_user_func_array($matcher, [$pattern, substr($from, $endOfMatch), &$matches])) {
```

Comment:
I wrapped this in call_user_func_array to support preg_match if mb_ereg is not available